lib/scripts/update_agent_config.sh

```diff
@@ -1,13 +1,30 @@
 #!/bin/sh
 
 # wait for the buildpack scripts to finish
-timeout=0
-while { ! [ "$(pgrep -f ./agent)" = "" ] && ! [ "$(pgrep -f ./dogstatsd)" = "" ]; } && [ $timeout -lt 120 ]; do
```

Comment:
Since these processes are started using a pid file and it's logically possible for the associated buildpack applications to have a binary with the same name, does use of -F make more sense with pgrep?

Reply:
Yes, it does make more sense. However, when this script is invoked by the Garden API, the underlying process doesn't have the `DATADOG_DIR` ENV variable, which we would need to find the pid file (e.g.: `${DATADOG_DIR}/run/agent.pid`) . @sarah-witt was using this approach as a temporary workaround. We didn't want to assume that `DATADOG_DIR=/home/vcap/app/.datadog` without careful consideration, as we noticed in some parts in the scripts `DATADOG_DIR` could be a user input. 
We're going to make a separate PR to address this issue and then we'll resume this PR.